bodhi/tests/server/services/test_updates.py

```diff
@@ -275,6 +277,43 @@ def test_new_rpm_update_koji_error(self, *args):
         self.assertEqual(up['errors'][0]['description'],
                          "Koji error getting build: bodhi-2.0.0-2.fc17")
 
+    @mock.patch.dict('bodhi.server.validators.config', {'acl_system': 'dummy'})
+    @mock.patch(**mock_uuid4_version1)
+    @mock.patch(**mock_valid_requirements)
+    def test_new_rpm_update_from_tag(self, *args):
+        """Test creating an update using builds from a Koji tag."""
+        # We don't want the new update to obsolete the existing one.
+        self.db.delete(Update.query.one())
+
+        update = self.get_update(builds=None, from_tag='f17-updates-candidate')
```

Comment:
This fails in CI because the code does not expect builds to be `None`, only a list (that can be empty). I suggest to either use `[]` here or to handle in the validation code the case where `builds` is `None`.

Reply:
Where does it fail and how? AIUI, it uses `BaseTestCaseMixin.get_update()` which has this:

```
        if builds:
            if isinstance(builds, list):
                builds = ','.join(builds)
            if not isinstance(builds, str):
                builds = builds.encode('utf-8')
            update['builds'] = builds
```

I.e. if `builds` is set to `None`, it should simply be missing from the returned dictionary.

Reply:
In the CI logs you'll see:
```
13:55:12  Traceback (most recent call last):
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/tweens.py", line 41, in excview_tween
13:55:12      response = handler(request)
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/router.py", line 148, in handle_request
13:55:12      registry, request, context, context_iface, view_name
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/view.py", line 667, in _call_view
13:55:12      response = view_callable(context, request)
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/config/views.py", line 169, in __call__
13:55:12      return view(context, request)
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/config/views.py", line 188, in attr_view
13:55:12      return view(context, request)
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/config/views.py", line 214, in predicate_wrapper
13:55:12      return view(context, request)
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/viewderivers.py", line 384, in authdebug_view
13:55:12      return view(context, request)
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/viewderivers.py", line 325, in secured_view
13:55:12      return view(context, request)
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/viewderivers.py", line 436, in rendered_view
13:55:12      result = view(context, request)
13:55:12    File "/usr/lib/python3.7/site-packages/pyramid/viewderivers.py", line 144, in _requestonly_view
13:55:12      response = view(request)
13:55:12    File "/usr/lib/python3.7/site-packages/cornice/service.py", line 493, in wrapper
13:55:12      validator(request, **args)
13:55:12    File "/bodhi/bodhi/server/validators.py", line 90, in validator
13:55:12      f(request, **kwargs)
13:55:12    File "/bodhi/bodhi/server/validators.py", line 203, in validate_build_nvrs
13:55:12      for build in request.validated.get('builds', []):
13:55:12  TypeError: 'NoneType' object is not iterable
```